burst-coroutines/src/commonMain/kotlin/app/cash/burst/coroutines/CoroutineTestInterceptor.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (C) 2025 Cash App
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package app.cash.burst.coroutines
+
+interface CoroutineTestInterceptor {
```

Comment:
Rather than copy-paste the docs I left these bare for now. I’ll copy-paste them _carefully_ in follow-up